server/cmwell-dc/src/main/scala/cmwell/dc/stream/TsvRetriever.scala

```diff
@@ -179,7 +179,21 @@ object TsvRetriever extends LazyLogging {
                 val consumeCount = Settings.initialTsvRetryCount - state.retriesLeft + 1
                 yellowlog.info(s"TSV (bulk)consume succeeded only after $consumeCount (bulk)consumes. token: ${state.tsvRetrieveInput}.")
               }
-              Success(TsvFlowOutput(sortedData, nextPositionKey, s.intValue == 204)) -> state.copy(lastException = None)
+              //for bulk-consume, X-CM-WELL-N header exists. Check that the number of TSVs got is the same as the number in the header.
+              h.find(_.name == "X-CM-WELL-N").fold{
+                Try(TsvFlowOutput(sortedData, nextPositionKey, s.intValue == 204)) -> state.copy(lastException = None)
+              } { nHeader =>
+                val expectedTsvCount = nHeader.value().toInt
+                val gotTsvCount = sortedData.size
+                if (gotTsvCount != expectedTsvCount) {
+                  val e = new Exception(s"Got $gotTsvCount but the expected count (taken from X-CM-WELL-N header) is $expectedTsvCount.")
```

Comment:
you don't need `Exception` with stack trace here, since you use it as cause in a line below... 
it'll probably be better to use `NoStackTrace` trait instead

not very important though...